test/infrastructure/inmemory/internal/cloud/runtime/cache/client.go

```diff
@@ -409,11 +409,8 @@ func (c *cache) doTryDeleteLocked(resourceGroup string, tracker *resourceGroupTr
 		delete(tracker.ownedObjects, ownReference{gvk: objGVK, key: objKey})
 	}
 
-	// If the object still has finalizers, only set the deletion timestamp if not already set.
-	if len(obj.GetFinalizers()) > 0 {
-		if !obj.GetDeletionTimestamp().IsZero() {
-			return false, nil
-		}
+	// Set the deletion timestamp if not already set.
+	if obj.GetDeletionTimestamp().IsZero() {
```

Comment:
@fabriziopandini  - would be good to get your opinion on the changes to the delete flow. I couldn't get delete events in the informer before making this change. There's currently a unit test failing because it doesn't expect this to be recorded as an event.

Reply:
Just fyi. I'm using this code since a few days in the scale tests. So far everything worked (but I didn't too a lot of deletions)

Reply:
I would expect that we need this change to be able to send deletion events for objects without finalizers (Nodes ?!)

Reply:
I'm ok with this change.
The only downside is that we will send events to in-memory controllers for objects that could go away right after, but that's ok and we usually handle this as a first thing when we try to read the reconciled object